packages/@lwc/engine/src/framework/wc.ts

```diff
@@ -51,11 +51,20 @@ export function buildCustomElementConstructor(
         }
         connectedCallback() {
             const vm = getAssociatedVM(this);
-            appendRootVM(vm);
+            if (process.env.NODE_ENV !== 'production') {
+                assert.isTrue(
+                    vm.state === VMState.created || vm.state === VMState.disconnected,
```

Comment:
I don't think we care about this anymore since the browser does the right thing... probably can remove this block